include/deal.II/fe/fe_enriched.h

```diff
@@ -644,6 +648,299 @@ class FE_Enriched : public FiniteElement<dim, spacedim>
+  using predicate_function
+    = std::function< bool
+      (const typename Triangulation<dim,spacedim>::cell_iterator &) >;
+
+  /**
+   * ColorEnriched::Helper class creates a collection of FE_Enriched finite
+   * elements (hp::FECollection) to be used with hp::DoFHandler in a domain
+   * with multiple, possibly overlapping, sub-domains with individual
+   * enrichment functions. Note that the overlapping regions may have
+   * multiple enrichment functions associated with it. This is implemented
+   * using a general constructor of FE_Enriched object which allows different
+   * enrichment functions.
+   *
+   * To ensure $C^0$ continuity when an enriched sub-domain
+   * (characterized by a single enrichment function) shares
+   * an interface only with non-enriched FE, we can
+   * use a FE_Enriched object in enriched
+   * sub-domain and in non-enriched domain
+   * a standard FE (eg: FE_Q) wrapped into an FE_Enriched
+   * object (which internally uses dominating FE_Nothing object).
+   * Refer to documentation on FE_Enriched for more
+   * information on this. The situation becomes more
+   * complicated when two enriched sub-domains
+   * share an interface. When the number of enrichment functions are
+   * different for the two sub-domains, it is handled by using
+   * dominating FE_Nothing object (a dummy enriched FiniteElement).
+   * This is similar to how interface between enriched and non-enriched
+   * domains is handled. It is ensured that both sub-domains have same number of
+   * enrichment functions (dummy or not). Another complication is when
+   * they have different (kind of) enrichment functions
+   * (Say the corresponding FE_Enriched objects are fe_enr1, fe_enr2).
+   * This is handled by pairing them with a dominating FE_Nothing i.e
+   * use FE_Enriched<dim>[fe_base, fe_enr1, fe_nothing] and
+   * FE_Enriched<dim>[fe_base, fe_nothing, fe_enr2]. Note that
+   * size of vector of enriched FiniteElements is equal to 2, the
+   * same as number of enrichment functions. Using the above two
+   * strategies, we construct the appropriate FE_Enriched
+   * using the general constructor. Note that this is
+   * done on mesh without hanging nodes.
+   *
+   * Consider a problem with multiple sub-domains, each with a
+   * different enrichment function. As discussed previously,
+   * this will require that number of enriched FiniteElements for
+   * constructing the FE_Enriched object
+   * be the same as number of enrichment functions, possibly
+   * using a number of dummy FE_Nothing objects. This is because
+   * we are not using the information of how the domains are connected
+   * and any sub-domain may share interface with another (Not
+   * considering overlaps for now!).
+   * However, in general, a given sub-domain
+   * shares interface only with a few sub-domains. This warrants
+   * the use graph coloring algorithm to reduce
+   * the size of vector of enriched FiniteElements
```

Comment:
the size of the vector

By the way, I would avoid using *FiniteElements* here and elsewhere and instead write *FiniteElement objects* or *finite elements*. The first alternative will insert the link to the `FiniteElement` class, whereas your spelling will just end up looking a bit weird.